[WIP] Post-RFC0035 cleanup

Synthetica9 opened the pull request:
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Synthetica9 committed:
treewide: remove unneeded name = "${pname}-${version}"

volth commented on pkgs/applications/networking/feedreaders/feedreader/default.nix:
```diff
@@ -7,7 +7,7 @@ let
   version = "2.2";
 in stdenv.mkDerivation {
-  name = "${pname}-${version}";
+  inherit pname version;
```
It case like this one, both ```let```/```in``` and ```inherit``` are not needed (although sometimes it would require ```rec```)

Synthetica9 replied:
Yeah, I just `sed`ded for this round, but I'll do some more manual review.

Synthetica9 replied:
... Actually, I'll guess I'll extend [nix-linter](https://github.com/synthetica9/nix-linter) to do that review for me!

Synthetica9 committed:
treewide: remove unneeded let-inherit pairs

Found by (shameless plug) nix-linter
(https://github.com/synthetica9/nix-linter)

mmahut commented:
Are there any updates on this pull request, please?
